main.go

```diff
@@ -305,8 +307,25 @@ func split(s string) (head, tail string) {
 func main() {
 	defer task.Exit()
 
-	if len(os.Args) > 1 {
-		task.StartNonInteractive()
+	interactive := flag.Bool("i", true, "enable interactive mode")
+	command := flag.String("c", "", "parse next argument instead of stdin")
+
+	flag.Parse()
+
+	if *command != "" {
+		task.StartNonInteractive(*command, flag.Args())
```

Comment:
What if *interactive is true?

Reply:
Command takes preference. Here is the current behavior:

```
oh$ oh -c
flag needs an argument: -c
Usage of oh:
  -c string
        parse next argument instead of stdin
  -i    enable interactive mode (default true)
oh$ oh -i -c
flag needs an argument: -c
Usage of oh:
  -c string
        parse next argument instead of stdin
  -i    enable interactive mode (default true)
oh$ oh -i -c "echo Hello"
Hello
oh$ oh -c "echo Hello"
Hello
oh$ oh -c "echo Hello" -i
Hello
oh$ oh -c "echo Hello" -i=false
Hello
oh$ oh -i=false
Non interactive session needs either a command (-c) or a file as argument
Usage of oh:
  -c string
        parse next argument instead of stdin
  -i    enable interactive mode (default true)
oh$ oh /tmp/test.oh
oh$ 
```

Reply:
Something that do not work in this version or master is:
```
oh$ cat /tmp/test
echo "Hello World"
oh$ oh /tmp/test
oh$
```

should't the string be printed?

Reply:
That works on the master branch:
```
oh$ git branch
  devel
* master
oh$ cat /tmp/test
echo "Hello World"
oh$ oh /tmp/test
Hello World
```

Reply:
I got my oh versions mixed up, that's why I didn't see that. Seems that I also expected flags.Args() to return the same array of os.Args when there are no flags, but it didn't work. I have fixed that and the file tuns as expected.